### ISSUE TITLE
Cannot load module: does not provide an export named 'SimpleEventDispatcher'

### ISSUE THREAD

**Issue by emielvangoor** [KeesCBakker/Strongly-Typed-Events-for-TypeScript#251]
Hi all!

I've got a problem when loading the package in my Typescript application. I've installed the package using yarn:
`yarn add strongly-typed-events`.

When i run my application i get the following error in the console of the browser.

```javascript
Uncaught SyntaxError: The requested module './../../../node_modules/strongly-typed-events/dist/index.js' does not provide an export named 'SimpleEventDispatcher'
```

I think it has something to do with the way Typescripts loads the libraries? (i've got limit knowledge on ts.)

I've got the following `tsconfig.json`:

```json
{
  "compilerOptions": {
    "target": "es2018",
    "module": "esnext",
    "moduleResolution": "node",
    "noEmitOnError": true,
    "lib": ["es2017", "dom"],
    "strict": true,
    "esModuleInterop": false,
    "outDir": "out-tsc",
    "rootDir": "./",
    "experimentalDecorators": true,
    "resolveJsonModule": true,
    "allowSyntheticDefaultImports": true
  },
  "include": ["./src/**/*.ts"]
}
```

in my application i load the `SimpleEventDispatcher` by importing the library: 

```javascript
import { SimpleEventDispatcher } from 'strongly-typed-events';
``` 
The Typescript compiler does not complain about this, it's only when i run the application in the browser i get the message.

What am i missing?

**Comment by KeesCBakker** [comment]
What version of the package are you using.

**Comment by emielvangoor** [comment]
Hi Kees,

Sorry i didn't mention it, i'm using version `"strongly-typed-events": "^2.1.8",`

**Comment by KeesCBakker** [comment]
Does the latest version suffer from the same problem?

**Comment by KeesCBakker** [comment]
If I look at this one, it works: https://replit.com/@KeesCC/DistinctOtherFunction#index.ts. I've tried it with `"strongly-typed-events": "^2.1.8"`. Might it have something to do with the `tsconfig` property `module`? When I switch it to `esnext` it looks like it fails, but not with the same error.

Can you send me a small setup of this error?

**Comment by riffy** [comment]
Hello, sorry for jumpin into this issue. I encountered the same issue a while ago (~ 6months). I remember that two things were causing this:
1. `tsconfig.json` with `"module": "esnext"` and `"moduleResolution": "node"`
2. `package.json` with `"type": "module"`

But those properties were necessary in my case, since I am developing a 'plugin' for a system (GTA V alt-V Multiplayer) using this neat lib.

I circumvented this issue, by manually creating one `strongly-typed-events.ts` based on the release (and replaced some `setTimeout` with other built-in functiopns from the system) with `"use strict"` and using the supported export/import.

Maybe this information helps a little bit...

**Comment by KeesCBakker** [comment]
@emielvangoor does the setup of @riffy look a bit like the setup in which you encountered the problem?

**Comment by emielvangoor** [comment]
> @emielvangoor does the setup of @riffy look a bit like the setup in which you encountered the problem?

It does. My `tsconfig.json` is the same as @riffy with `"module": "esnext"` and `"moduleResolution": "node"` but i don't have a `"type": "module"` in my `package.json`.

**Comment by KeesCBakker** [comment]
So it narrows probably down to  `"module": "esnext"` and `"moduleResolution": "node"` 🤔

**Comment by riffy** [comment]
@emielvangoor I just tried to recreate the issue with your `tsconfig.json`, but I could not 100% recreate it, because my error message now shows (but I remember I had the same issue as you)
```
(node:19800) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
D:\ste-issue\out-tsc\src\app.js:1
import { SignalDispatcher, SimpleEventDispatcher, EventDispatcher } from "strongly-typed-events";
^^^^^^
```

If you are allowed to change the `tsconfig` in your project, you could try to change the [module ](https://www.typescriptlang.org/docs/handbook/modules.html)to `"CommonJS"` as `require` is then used instead of `import`. (I wasn't)

Or add the `"type": "module"` to `package.json`.
I had this issue with some other packages too, and some others worked perfectly fine out of the box (this is the part i never understood 🙄 )

@KeesCBakker 
I think (!) its affected by how some libraries, e.g. ste, are built. I had similiar problems with `unique-names-generator` and they also use `commonjs` module in the tsconfig, while the ones that seemed to work seemingly perfect were built around `es2020` or `esnext` 
Sorry I'm no expert when it comes to build-/compileoptions, I'm just doing whatever works :octocat:

**Comment by KeesCBakker** [comment]
I'm not sure what we can do to fix this problem. I love to do a major version update if we have a fix.

**Comment by riffy** [comment]
Its no necessarily a problem of your code, but rather a misbehavior of config and buildoptions and the different ecmascript versions. I will invest some time at the weekend and make a PR if I find a possible solution

**Comment by KeesCBakker** [comment]
I think the issue is related to: https://github.com/KeesCBakker/Strongly-Typed-Events-for-TypeScript/issues/269

**Comment by KeesCBakker** [comment]
Could be fixed now since version 3.0.1, can you please check?

**Comment by KeesCBakker** [comment]
I'm closing the issue due to inactivity.
